### PR TITLE
Avoid overriding backend auth responses

### LINKED ISSUE #1233: MGW returns mgw 401 message when backend returns a 401 response
### Description:

When the backend endpoint respond with 401 error message. Microgateway sends its own 401 error to the client.
```
{"fault":{"code":900901, "message":"Invalid Credentials", "description":"Invalid Credentials. Make sure you have given the correct access token"}}
```

### Steps to reproduce:
1. Send a request that is valid for mgw but 401 for backend

### Affected Product Version:

3.1.0

### Environment details (with versions):
- OS:
- Client:
- Env (Docker/K8s):

### PR DESCRIPTION
**Pull request by praminda**
### Purpose

Avoid MGW from sending its own auth failure responses when backends and interceptors respond with auth failures.
### Issues

Fixes #1233

### Automation tests
 - Unit tests added: No
 - Integration tests added: Yes

### Tested environments

MacOS

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: tests/src/test/java/org/wso2/micro/gateway/tests/endpoints/EndpointWithSecurityTestCase.java [hunk 1]
```diff
@@ -52,6 +52,31 @@ public void testPerAPISecureEndpoint() throws Exception {
         Assert.assertEquals(response.getResponseCode(), 200, "Response code mismatched");
     }
 
+    @Test(description = "Test Invoking the resource which endpoint defined at API level using references")
```
Comment (Rajith90) [inline tests/src/test/java/org/wso2/micro/gateway/tests/endpoints/EndpointWithSecurityTestCase.java:55]: Lets change the description properly

File: tests/src/test/java/org/wso2/micro/gateway/tests/endpoints/EndpointWithSecurityTestCase.java [hunk 2]
```diff
@@ -52,6 +52,31 @@ public void testPerAPISecureEndpoint() throws Exception {
...
+        headers.put(HttpHeaderNames.AUTHORIZATION.toString(), "Bearer " + jwtTokenProd);
+        org.wso2.micro.gateway.tests.util.HttpResponse response = HttpClientRequest
+                .doGet(getServiceURLHttp("petstore/v5/store/order/1"), headers);
+        Assert.assertNotNull(response);
+        Assert.assertEquals(response.getData(), ResponseConstants.AUTHENTICATION_FAILURE_RESPONSE);
+        Assert.assertEquals(response.getResponseCode(), 401, "Response code mismatched");
+    }
+
+
+    @Test(description = "Test Invoking the resource which endpoint defined at API level using references")
```
Comment (Rajith90) [inline tests/src/test/java/org/wso2/micro/gateway/tests/endpoints/EndpointWithSecurityTestCase.java:68]: Lets change the description properly

File: tests/src/test/java/org/wso2/micro/gateway/tests/endpoints/EndpointWithSecurityTestCase.java (hunk 1 above)
Comment (praminda) [reply tests/src/test/java/org/wso2/micro/gateway/tests/endpoints/EndpointWithSecurityTestCase.java]: done

File: tests/src/test/java/org/wso2/micro/gateway/tests/endpoints/EndpointWithSecurityTestCase.java (hunk 2 above)
Comment (praminda) [reply tests/src/test/java/org/wso2/micro/gateway/tests/endpoints/EndpointWithSecurityTestCase.java]: done

**Pull request merged**